### ISSUE TITLE
smallrye-jwt not working in native mode

### ISSUE THREAD

**Issue by starksm64** [quarkusio/quarkus#1163]
Testing the using-jwt-rbac under native mode produces no errors, but neither authentication or authorization are working.

**Comment by sebastienblanc** [comment]
@starksm64 Hi Scott ! Is there any (even dirty :) ) workaround for this right now ? I would like to demo this in 10 days at a conference and it would be nice if I could show the native image working as well. Thx !

**Comment by starksm64** [comment]
I'll push my local changes to my quarkus fork so that you could take and build that for a demo in the interim. I'll post another comment here when it is pushed.

**Comment by starksm64** [comment]
I have pushed my changes to a new public repo on the iss1163 branch:
https://github.com/starksm64/quarkus/tree/iss1163

With that, the using-jwt-rbac quickstart integration tests run under native mode, although there are a couple of fixes in that as well. One is a bug in the testHelloDenyAll() test a wrong annotation on the associated endpoint.

The other is that in native mode, the LottoNumbersResource is not being introspected correctly due to what looks like a bug in the resteasy extension, but can be worked around by adding the `@RegisterForReflection` annotation to the class:

```java
import io.quarkus.runtime.annotations.RegisterForReflection;
import org.eclipse.microprofile.jwt.Claim;
import org.eclipse.microprofile.jwt.Claims;
import org.eclipse.microprofile.jwt.JsonWebToken;

@RegisterForReflection
@Dependent
public class LottoNumbersResource {
    @Inject
    JsonWebToken jwt;
    @Inject
    @Claim(standard = Claims.birthdate)
    Optional<JsonString> birthdate;
...
```

**Comment by sebastienblanc** [comment]
@starksm64 Thanks ! I will give it a try tomorrow !

**Comment by sebastienblanc** [comment]
@starksm64 Works perfectly !

**Comment by sebastienblanc** [comment]
Is this issue planned to be fixed in 0.13 ?

**Comment by starksm64** [comment]
Yes, either via the workaround that I have or via the more general provider registration that I'm testing.
